apps/meteor/ee/server/patches/mergeContacts.ts

```diff
@@ -21,28 +23,28 @@ export const runMergeContacts = async (
 	if (!channel) {
 		throw new Error('error-invalid-channel');
 	}
-	const similarContacts: ILivechatContact[] = await LivechatContacts.findSimilarVerifiedContacts(channel, contactId);
+	const similarContacts: ILivechatContact[] = await LivechatContacts.findSimilarVerifiedContacts(channel, contactId, { session });
 
 	if (!similarContacts.length) {
 		return originalContact;
 	}
 
 	for await (const similarContact of similarContacts) {
-		const fields = await ContactMerger.getAllFieldsFromContact(similarContact);
-		await ContactMerger.mergeFieldsIntoContact(fields, originalContact);
+		const fields = ContactMerger.getAllFieldsFromContact(similarContact);
+		await ContactMerger.mergeFieldsIntoContact(fields, originalContact, undefined, session);
 	}
 
 	const similarContactIds = similarContacts.map((c) => c._id);
-	const { deletedCount } = await LivechatContacts.deleteMany({ _id: { $in: similarContactIds } });
+	const { deletedCount } = await LivechatContacts.deleteMany({ _id: { $in: similarContactIds } }, { session });
 	logger.info(
 		`${deletedCount} contacts (ids: ${JSON.stringify(similarContactIds)}) have been deleted and merged with contact with id ${
 			originalContact._id
 		}`,
 	);
 
-	await LivechatRooms.updateMany({ 'v.contactId': { $in: similarContactIds } }, { $set: { 'v.contactId': contactId } });
+	await LivechatRooms.updateMany({ 'v.contactId': { $in: similarContactIds } }, { $set: { 'v.contactId': contactId } }, { session });
```

Comment:
Do this in a model function

Reply:
hey could you add this comment in the base branch?

Reply:
Won't that cause unnecesary conflicts to you? 👀 like, the base branch update, then you having to update the model typings there and so on.

I can add, but would like to avoid the double work for u 🤗

Reply:
I was talking with the team to see if this task was necessary to be merged asap, but we did manage to get a fully working version on the feature branch, so there is no need to hurry and I can do this in my branch :P